queries.sql

```diff
@@ -55,4 +55,91 @@ GROUP BY neutered;
 
 SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
 
-SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth >= '01-01-1990' AND date_of_birth < '12-31-2000' GROUP BY species;
\ No newline at end of file
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth >= '01-01-1990' AND date_of_birth < '12-31-2000' GROUP BY species;
+
+
+SELECT animals.name FROM animals 
+JOIN owners ON animals.owners_id = owners.id 
+WHERE owners.full_name = 'Melody Pond';
+
+SELECT animals.*, species.name AS species_name FROM animals 
+JOIN species ON animals.species_id = species.id WHERE species.name = 'Pokemon';
+
+SELECT animals.name AS animal_name, owners.fulL_name AS owner_name FROM owners 
+LEFT JOIN animals ON owners.id = animals.owners_id;
+
+SELECT species.name, COUNT(*) FROM animals 
+JOIN species ON species.id = animals.species_id GROUP BY species.name;
+
+SELECT animals.name AS animal_name, owners.full_name AS owner_name, species.name AS species_name FROM animals
+JOIN species ON species.id = animals.species_id JOIN owners ON owners.id = animals.owners_id WHERE species.name = 'Digimon' AND owners.full_name = 'Jennifer Orwell';
+
+SELECT * FROM animals
+JOIN owners ON animals.owners_id = owners.id 
+WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
+
+SELECT owners.full_name, COUNT(animals.name) AS num_animals FROM owners
+LEFT JOIN animals ON owners.id = animals.owners_id GROUP BY owners.full_name ORDER BY num_animals DESC;
+
+--visits
+
+--William Thatcher last visit
+
+SELECT vet.name AS vet_name, an.name AS animal_name
+  FROM visits v JOIN animals an ON an.id = v.animal_id
+  JOIN vets vet ON vet.id = v.vet_id
+  WHERE vet.name = 'William Tatcher'
+  ORDER BY v.date_of_visit DESC LIMIT 1;
+
+-- Stephanie Mendez  animal visits
+SELECT vet.name as vet_name, COUNT(date_of_visit) 
+  FROM visits v JOIN vets vet ON v.vet_id = vet.id
+  WHERE name = 'Stephanie Mendez' GROUP BY vet.name;
+
+  --all vets and specialties
+
+  SELECT 
+	vets.name,
+	species.name AS specialized_in
+FROM vets
+LEFT JOIN specializations ON specializations.vets_id = vets.id
+LEFT JOIN  species ON specializations.species_id = species.id;
+
+--Stephanie Mendez 2020-04-01-2020-08-30
+SELECT animals.name AS animal_name, visits.date_of_visit FROM animals
+JOIN visits ON visits.animals_id = animals.id
+JOIN vets ON vets.id = visits.vets_id
```

Comment:
- Kindly check the `vet_id` and the `animal_id` columns on the `visits` table. Please, change the above ☝️ from  `visits.animals_id` to `visits.animal_id` and from `visits.vets_id` to `visits.vet_id`. 😉